File: src/main/java--/com/aaronhowser1/pitchperfect/packets/ModPacket.java
package com.aaronhowser1.pitchperfect.packets;

import net.minecraft.network.FriendlyByteBuf;
import net.minecraftforge.network.NetworkEvent;

import java.util.function.Supplier;

public interface ModPacket {

    void encode(FriendlyByteBuf buffer);
    void receiveMessage(Supplier<NetworkEvent.Context> context);
}

File: src/main/kotlin/com/aaronhowser/pitchperfect/packet/ModPacket.kt
package com.aaronhowser.pitchperfect.packet

import net.minecraft.network.FriendlyByteBuf
import net.minecraftforge.network.NetworkEvent
import java.util.function.Supplier

interface ModPacket {

    fun encode(buffer: FriendlyByteBuf)
    fun receiveMessage(context: Supplier<NetworkEvent.Context>)

}